jwst/datamodels/container.py

```diff
@@ -87,13 +90,16 @@ def __init__(self, init=None, persist=True, **kwargs):
             self._ctx = self
             self.__class__ = init.__class__
             self._models = init._models
+        elif is_association(init):
+            self.from_asn(init)
         elif isinstance(init, str):
             try:
-                self.from_asn(init, **kwargs)
+                init_from_asn = self.read_asn(init)
             except (IOError):
                 raise IOError('Cannot open files.')
```

Comment:
Instead of triggering off the `IOError` and then re-emitting another `IOError` could we just raise the existing `IOError` along with its error message?  It would avoid having then "While handling error X another error Y was..." messages.  Same below.  I know these are not your changes, but it would clean up the errors a bit.